azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -92,9 +95,11 @@
 
    private final AzureComputeConstants azureComputeConstants;
 
+   private final ProviderMetadata providerMetadata;
+
    @Inject
    AzureComputeServiceAdapter(final AzureComputeApi api, final AzureComputeConstants azureComputeConstants,
-                              CleanupResources cleanupResources) {
+                              CleanupResources cleanupResources, ProviderMetadata providerMetadata) {
```

Comment:
I think in this way the PROPERTY_ZONES becomes mandatory (otherwise Guice is not happy)
Is it what we want?